Data/Aeson/Parser/Internal.hs

```diff
@@ -390,12 +392,25 @@ eitherDecodeWith p to s =
       L.Done _ v     -> case to v of
                           ISuccess a      -> Right a
                           IError path msg -> Left (path, msg)
-      L.Fail _ ctx msg -> Left ([], buildMsg ctx msg)
+      L.Fail notparsed ctx msg -> Left ([], buildMsg notparsed ctx msg)
   where
-    buildMsg :: [String] -> String -> String
-    buildMsg [] msg = msg
-    buildMsg (expectation:_) msg =
-      msg ++ ". Expecting " ++ expectation
+    buildMsg :: L.ByteString -> [String] -> String -> String
+    buildMsg notParsed [] msg = msg ++ grabErrorLine notParsed
+    buildMsg notparsed (expectation:_) msg =
+      msg ++ ". Expecting " ++ expectation ++ grabErrorLine notparsed
+    grabErrorLine :: L.ByteString -> String
+    grabErrorLine bs =
+        let notParsedStr =
+                C.unpack $
+                 -- if newline is present cut at that position
+                BSL.takeWhile (10 /=)
+                 -- take 100 bytes
+                 (BSL.take 100 $
```

Comment:
`BSL.take 100` should be the first thing done. I'm *not sure*, but if one can make parser fail before "very large" whitespace block, the `dropWhile` can do a lot of work.

OTOH, if you have 100 of space in your JSON document, that's pretty easy to spot.